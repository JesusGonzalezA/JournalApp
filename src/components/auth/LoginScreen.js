import React from 'react'
import { useDispatch } from 'react-redux'

import { Link } from 'react-router-dom'
import { login, startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [ formValues, handleInputChange ] = useForm({
        email: '',
        password: ''
    })
    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch( startLoginEmailPassword(email, password) );
    }

    const handleGoogleLogin = (e) => {
        dispatch( startGoogleLogin() );
    }
    
    return (
        <>
            <h1 className="auth__title mb-5">Login</h1>

            <form onSubmit={ handleLogin }>
                <input 
                    type="text"
                    placeholder="E-mail"
                    name="email"
                    className="auth__input"
                    value={email}
                    onChange={ handleInputChange }                    
                />

                <input  
                    type="password"
                    placeholder="Password"
                    name="password"
                    className="auth__input"
                    password={password}
                    onChange={ handleInputChange }
                />

                <button type="submit" className="btn btn-primary btn-block">
                    Login
                </button>
               
                <div className="auth__social-networks">
                    <p>Login with your fav social network</p>

                    <div 
                        className="google-btn"
                        onClick={ handleGoogleLogin }
                    >
                        <div className="google-icon-wrapper">
                            <img 
                                className="google-icon" 
                                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" 
                                alt="Sign in with Google" 
                            />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with Google</b>
                        </p>
                    </div>
                </div>
                
                <Link to="/auth/register" className="link">
                    Not registered yet?
                </Link>

            </form>
        </>
    )
}
