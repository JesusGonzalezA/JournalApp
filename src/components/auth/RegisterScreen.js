import React from 'react'
import validator from 'validator'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import { useDispatch, useSelector } from 'react-redux'
import { removeError, setError } from '../../actions/ui'
import { startRegisterWithEmailPasswordName } from '../../actions/auth'

export const RegisterScreen = () => {

    const dispatch = useDispatch();
    const { msgError } = useSelector( state => state.ui );

    const [ formValues, handleInputChange ] = useForm({
        email: '',
        name: '',
        password: '',
        password2: ''
    });
    
    const {
        email,
        name,
        password,
        password2
    } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();

        if ( isFormValid() ){
            dispatch ( startRegisterWithEmailPasswordName(email, password, name) )
        }
    }

    const isFormValid = () => {

        if ( !validator.isEmail( email ) ){
            dispatch( setError('Email is not valid') );
            return false;
        }
        if ( name.trim().length === 0 ){
            dispatch( setError('Name is required') );
            return false;
        }
        if ( password !== password2){
            dispatch( setError('Password should match') );
            return false;
        }
        if ( password.length < 5 ){
            dispatch( setError('Password should be at least 6 char long') );
            return false;
        }

        dispatch( removeError() );
        return true;
    }

    return (
        <>
            <h1 className="auth__title mb-5">Sign up</h1>

            <form 
                className="animate__animated animate__fadeIn animate__faster"
                onSubmit={ handleRegister }
            >

                {
                    msgError && 
                    <div className="auth__alert-error">
                        { msgError }
                    </div>
                }

                <input 
                    type="text"
                    placeholder="E-mail"
                    name="email"
                    value={email}
                    className="auth__input"
                    onChange={ handleInputChange }
                />

                <input 
                    type="text"
                    placeholder="Name"
                    name="name"
                    value={name}
                    className="auth__input"
                    onChange={ handleInputChange }
                />

                <input  
                    type="password"
                    placeholder="Password"
                    name="password"
                    value={password}
                    className="auth__input"
                    onChange={ handleInputChange }
                />


                <input  
                    type="password"
                    placeholder="Cofirm password"
                    name="password2"
                    value={password2}
                    className="auth__input"
                    onChange={ handleInputChange }
                />

                <button type="submit" className="btn btn-primary btn-block mb-5">
                    Sign up
                </button>
               
                <Link to="/auth/login" className="link">
                    Already registered?
                </Link>

            </form>
        </>
    )
}
