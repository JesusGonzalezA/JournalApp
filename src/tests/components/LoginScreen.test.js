import React from 'react';
import { mount } from 'enzyme'
import { LoginScreen } from '../../components/auth/LoginScreen';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const inititalState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};
let store = mockStore( inititalState )

const wrapper = mount( 
    <Provider store={store}>
        <MemoryRouter>
            <LoginScreen />
        </MemoryRouter>
    </Provider>
);

describe('Testing LoginScreen Component', () => {
    
    beforeEach(() => {
        store = mockStore( inititalState )
    });

    test('should match the snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})
