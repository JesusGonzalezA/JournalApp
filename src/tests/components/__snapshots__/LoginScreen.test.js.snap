// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen Component should match the snapshot 1`] = `
Array [
  <h1
    className="auth__title mb-5"
  >
    Login
  </h1>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="E-mail"
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      password=""
      placeholder="Password"
      type="password"
    />
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <div
      className="auth__social-networks"
    >
      <p>
        Login with your fav social network
      </p>
      <div
        className="google-btn"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="Sign in with Google"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with Google
          </b>
        </p>
      </div>
    </div>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Not registered yet?
    </a>
  </form>,
]
`;
